app/services/foreman/deprecation.rb

```diff
@@ -0,0 +1,21 @@
+module Foreman
+  class Deprecation
+    #aviliable options: controller, method, general info, callstack
+    def self.deprecation_warning(foreman_version_deadline, options={})
+      raise Foreman::Exception, _("Invalid version format, please enter in x.y.z") unless foreman_version_deadline.to_s.match(/\A^\d[.]\d+[.]\d$\z/)
```

Comment:
Hm okay, not sure that matters a lot.

1. The regexp now contains both \A and ^, ditto for EOL, just use one or the other.